### DISCUSSION TITLE
@typescript-eslint/typedef no longer supports allowExpressions option?

### DISCUSSION THREAD

**Discussion by antonioOrtiz** [typescript-eslint/typescript-eslint#7283] [Technical Discussions]
### Body

I'm getting a premptive popup in vscode (lower right) and an error when I try to commit my code:

![2023-07-20 11_08_20-package json - documentengine-service (Workspace) - Visual Studio Code](https://github.com/eslint/eslint/assets/4282013/bcfa9847-ce9c-4772-9859-1aab654f31c4)

What should about this rule with my current dependencies? Is there a workaround?

```
`package.json`
{
    "devDependencies": 
    ....other dependencies
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "eslint": "^8.45.0",
    ....other dependencies
}
```

```
.eslintrc.js
    '@typescript-eslint/typedef': [
    'error',
    {
    allowExpressions: true,
    variableDeclaration: true,
    arrowParameter: false
    }
],
```

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per our contributing guides - we do not use discussions for support.

Please see our contributing guide for alternatives.

https://typescript-eslint.io/contributing/discussions
